EIPS/eip-7509.md

```diff
@@ -0,0 +1,309 @@
+     * @param _componentAddress is the component address.
+     * @return true is attached, false is not attached
+     */
+    function hasComponent(uint256 _entityId, address _componentAddress)
+        external
+        view
+        returns (bool);
+
+    /**
+     * Remove a component from the entity.
+     * @dev Entity MUST be available.The component MUST have been added to the entity before.
+     * @param _entityId is the Id of the entity.
+     * @param _componentAddress is the address of the component to be removed.
+     */
+    function removeComponent(uint256 _entityId, address _componentAddress)
+        external;
+
+    /**
+     * Get the contract addresses of all components attached to the entity.
+     * @dev Entity MUST exist.
+     * @param _entityId is the Id of the entity.
+     * @return An array of contract addresses of the components owned by this entity.
+     */
+    function getEntityComponents(uint256 _entityId)
+        external
+        view
+        returns (address[] memory);
+
+    /**
+     * Register a system to the world.
+     * @dev MUST NOT register the same system to the world repeatedly.The system state is true means available, false means unavailable.
+     * @param _systemAddress is the contract address of the system.
+     * @param _systemState is the state of the system.
+     */
+    function registerSystem(address _systemAddress, bool _systemState) external;
+
+    /**
+     * Does the system exist in the world.
+     * @param _systemAddress is the contract address of the system.
+     * @return true exists, false does not exist.
+     */
+    function systemExists(address _systemAddress) external view returns (bool);
+
+    /**
+     * Get the contract addresses of all systems registered in the world.
+     * @return Array of contract addresses.
+     */
+    function getSystems() external view returns (address[] memory);
+
+    /**
+     * Set the system State.
+     * @dev System MUST exist.
+     * @param _systemAddress is the contract address of the system.
+     * @param _systemState is the state of the system.
+     */
+    function setSystemState(address _systemAddress, bool _systemState) external;
+
+    /**
+     * Get the state of a system.
+     * @param _systemAddress is the contract address of the system.
+     * @return The state of the system.
+     */
+    function getSystemState(address _systemAddress)
+        external
+        view
+        returns (bool);
+}
+```
+
+**IComponent.sol**
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity >=0.8.0;
+import "./Types.sol";
+
+interface IComponent {
+    /**
+     * The world contract address registered by the component.
+     * @return world contract address.
+     */
+    function world() external view returns (address);
+
+    /**
+     *Get the data type and get() parameter type of the component
+     * @dev SHOULD Import Types Library, which is an enumeration Library containing all data types.
```

Comment:
Since types from `Types.sol` appear in the signatures of functions, that library should be part of the Specification section. I might suggest a `### Library` heading above `### Interfaces` and having a paragraph like:

```markdown
### Library

The library [`Types.sol`](../assets/eip-7509/Types.sol) contains an enumeration of Solidity types used in the below interfaces.
```

Please note that we don't generally allow implementation details outside the Reference Implementation, but in this case, a library makes sense.